src/components/Dropdown/Dropdown.tsx

```diff
@@ -0,0 +1,107 @@
+import React, { ReactElement } from "react";
+import styled, { css } from "styled-components";
+import { Menu } from "@headlessui/react";
+import { IconDirection } from "../Icon/Direction";
+import { Text } from "../../typography";
+import { Button } from "../Button";
+import Icon from "../Icon";
+
+export interface DropdownOption {
+  value: string;
+  label: string;
+}
+
+export interface DropdownProps {
+  options: DropdownOption[];
+  defaultOption: DropdownOption;
+  selected?: DropdownOption;
+  setSelected: (option: DropdownOption) => void;
+}
+
+export const Dropdown = ({
+  options,
+  defaultOption,
+  selected,
+  setSelected,
+}: DropdownProps): ReactElement => {
+  const isDropdownOptionSelected = ({ value }: DropdownOption): boolean => {
+    return value === selected?.value;
+  };
+
+  return (
+    <Menu>
+      {({ open }) => (
+        <>
+          <StyledMenuButton>
+            <StyledLabel>{selected?.label || defaultOption.label}</StyledLabel>
+            <ChevronComponent open={open} />
+          </StyledMenuButton>
+          <StyledMenuItems>
+            {options.map((option) => (
+              <Menu.Item key={option.value} onClick={() => setSelected(option)}>
+                {({ active }) => (
+                  <StyledMenuItem
+                    active={active || isDropdownOptionSelected(option)}
+                  >
+                    {option.label}
+                  </StyledMenuItem>
+                )}
+              </Menu.Item>
+            ))}
+          </StyledMenuItems>
+        </>
+      )}
+    </Menu>
+  );
+};
+
+const ChevronComponent = ({ open }: { open: boolean }): ReactElement => {
+  const direction = open ? IconDirection.Up : IconDirection.Down;
+
+  return (
+    <Button.Icon icon={<Icon.Chevron direction={direction} size={16} />} />
```

Comment:
It could be nice to add a style on this button to support a smooth animation
```
    transition-property: transform;
    transition-timing-function: cubic-bezier(0.4, 0, 0.2, 1);
    transition-duration: 150ms;
```

This may require to change the rotate(180) by rotate(-180) on the buttons components but this is for another time!

Reply:
Rotating smoothly with https://github.com/Major-Eth/yearn-web-lib/pull/28/commits/02c1b91b397b53f5f6d567aae8b034e9e040e55e